firebase-vscode/src/workflow.ts

```diff
@@ -331,18 +337,58 @@ export async function setupWorkflow(
         success = await initHosting({
           spa: singleAppSupport,
           public: publicFolder,
-          useFrameworks: false
+          useFrameworks: false,
         });
       }
     }
     if (success) {
       readAndSendFirebaseConfigs(broker, context);
-      broker.send("notifyHostingInitDone",
-        { success, projectId, folderPath: currentOptions.cwd, framework: currentFramework });
+      broker.send("notifyHostingInitDone", {
+        success,
+        projectId,
+        folderPath: currentOptions.cwd,
+        framework: currentFramework,
+      });
       await fetchChannels(broker, true);
     } else {
-      broker.send("notifyHostingInitDone",
-        { success, projectId, folderPath: currentOptions.cwd });
+      broker.send("notifyHostingInitDone", {
+        success,
+        projectId,
+        folderPath: currentOptions.cwd,
+      });
+    }
+  }
+
+  // Opens a dialog prompting the user to select a directory.
+  // @returns string file path with directory location
+  async function selectDirectory() {
+    const selectedURI = await vscode.window.showOpenDialog({
+      canSelectFiles: false,
+      canSelectFolders: true,
+      canSelectMany: false,
+    });
+
+    /**
+     * If the user did not prematurely close the dialog and a directory in 
+     * which to put the new quickstart was selected, execute a sequence of
+     * shell commands that:
+     * 1. Downloads the quickstart into the selected directory with `git clone`
+     * 2. Enters the downloaded repo and deletes all unnecessary files and dirs
+     * 3. Moves all remaining files to the root of the selected directory
+     * 
+     * Once this download and configuration is complete, a new vscode window
+     * is opened to the selected directory.
+     */
+    if (selectedURI && selectedURI[0]) {
+      execSync(`git clone https://github.com/firebase/quickstart-js.git ` +
```

Comment:
Does the output of this show up in the terminal? If not, can you use `pluginLogger.info` to log it? If you don't have time to figure it out, add a todo and I can do it later. At least I think there should be a try/catch on this, that logs the error with `pluginLogger.error`.

Reply:
Gotcha; Wrapped the `execSync` call in a `pluginLogger.info` call added a try/catch block to log any errors in an upcoming commit.